### PR TITLE
Add hover/click events back to custom chat messages

### PR DESCRIPTION
**Pull request by lineargraph**
## Dependencies
- https://github.com/hannibal002/SkyHanni/pull/1512

## What
Describe what this pull request does, including technical details, screenshots, links to discord, etc.

<details>
<summary>Images</summary>

</details>

## Changelog Improvements
+ Added hover events back into the custom player message formatter. - nea

### CODE REVIEW

**Comment by hannibal002** [comment]
Please reformat the changelog

**Comment by lineargraph** [comment]
Does this work for a changelog?

File: src/main/java/at/hannibal2/skyhanni/data/hypixel/chat/event/GuildChatEvent.kt [hunk 1]
```diff
@@ -1,11 +1,15 @@
 package at.hannibal2.skyhanni.data.hypixel.chat.event
 
+import at.hannibal2.skyhanni.utils.ComponentSpan
 import net.minecraft.util.IChatComponent
 
+
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/data/hypixel/chat/event/GuildChatEvent.kt:6]: remove unnecessary empty lines

File: src/main/java/at/hannibal2/skyhanni/data/hypixel/chat/event/PlayerShowItemChatEvent.kt [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
-    val itemName: String,
+    val levelComponent: ComponentSpan?,
+    val action: ComponentSpan,
+    author: ComponentSpan,
+    val item: ComponentSpan,
+    message: ComponentSpan,
     chatComponent: IChatComponent,
     blockedReason: String? = null,
-) : AbstractChatEvent(author, message, chatComponent, blockedReason)
+) : AbstractChatEvent(author, message, chatComponent, blockedReason) {
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/data/hypixel/chat/event/PlayerShowItemChatEvent.kt]: remove {}

File: src/main/java/at/hannibal2/skyhanni/features/chat/playerchat/PlayerChatModifier.kt [hunk 3]
```diff
@@ -21,7 +21,8 @@ class PlayerChatModifier {
     fun onChat(event: SystemMessageEvent) {
         val newMessage = cutMessage(event.chatComponent.formattedText)
 
-        event.chatComponent = StringUtils.replaceIfNeeded(event.chatComponent, newMessage) ?: return
+        // TODO apply formattings correctly
+//         event.chatComponent = StringUtils.replaceIfNeeded(event.chatComponent, newMessage) ?: return
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/chat/playerchat/PlayerChatModifier.kt]: fix

File: src/main/java/at/hannibal2/skyhanni/utils/ComponentMatcherUtils.kt [hunk 4]
```diff
@@ -140,10 +163,11 @@ class ComponentSpan internal constructor(
     /**
      * Slice this component span. This is equivalent to the [String.substring] operation on the [text][getText].
      */
-    fun slice(start: Int, end: Int): ComponentSpan {
-        require(0 <= start)
-        require(start <= end)
-        require(end <= length)
+//     fun slice(start: Int = 0, end: Int = length - 1): ComponentSpan {
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/utils/ComponentMatcherUtils.kt]: remove comment

**Review by hannibal002 — CHANGES_REQUESTED** [review]
code cleanup and one fix

File: src/main/java/at/hannibal2/skyhanni/features/chat/playerchat/PlayerChatModifier.kt (hunk 3 above)
Comment (lineargraph) [reply src/main/java/at/hannibal2/skyhanni/features/chat/playerchat/PlayerChatModifier.kt]: Not sure what this was supposed to mean, tbh.

**Review by hannibal002 — APPROVED** [review]

**Review by hannibal002 — CHANGES_REQUESTED** [review]
chat formatting
image

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**